lib/properties.ex

```diff
@@ -106,12 +107,25 @@ defmodule PropCheck.Properties do
           tags = [[failing_prop: tag_property({module, name, []})]]
           prop_name = ExUnit.Case.register_test(__ENV__, :property, name, tags)
           def unquote(prop_name)(unquote(var)) do
+            module_default_opts = unquote(module_default_opts)
```

Comment:
I would suggest to put these cases blocks into an ordinary function, which is called here. This eases testing a lot, because you can test the functions independent of the macros. 

Also it is not clear for me why you differentiate between functions and lists. What about maps? They would work as well and are also used for configuring.

Reply:
Perhaps this helps also when searching for the root cause for the warning.

Reply:
Keywords or maps are both static configuration resolved on compilation time, function returns a config during run time.